controllers/addon_controller.go

```diff
@@ -154,7 +154,6 @@ func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request,
 			reason := fmt.Sprintf("Addon %s/%s could not be finalized. %v", instance.Namespace, instance.Name, err)
 			r.recorder.Event(instance, "Warning", "Failed", reason)
 			log.Error(err, "Failed to finalize addon.")
-			instance.SetInstallStatus(addonmgrv1alpha1.DeleteFailed, reason)
```

Comment:
as part of the refactor for `runWorkflow`, we are setting the install status and reason there; the wording for the reason might be different but I think that's ok since we're still logging the old wording here

Reply:
This is not using `runWorkflow` this is error during reconcile.

Reply:
ah good catch....`Finalize` has two paths for returning an err - one from `runWorkflow` and one from `Update` failure; I've added setting install status for the latter.

Reply:
Did you fix this?

Reply:
nvm, I see you did add status in `Finalize()` method.